tests/unit/validations/nested-model-relationship-test.js

```diff
@@ -390,53 +390,52 @@ module('Unit | Validations | Nested Model Relationships', function (hooks) {
 
   test('order with invalid question shows valid if invalid question is deleted in reverse order', function (assert) {
     assert.expect(9);
-    let order = run(() =>
-      this.owner
-        .lookup('service:store')
-        .createRecord('order', { id: 1, source: 'external' })
-    );
-
-    let orderLine,
-      orderSelection,
-      orderSelectionQuestion,
-      orderSelectionQuestion2;
-
     let store = this.owner.lookup('service:store');
-    run(() => {
-      let fakeSave = function (model) {
-        model.get('_internalModel').adapterWillCommit();
-        model.get('_internalModel').adapterDidCommit();
-      };
 
-      orderLine = store.createRecord('order-line', {
+    let payload = {
+      order: {
         id: 1,
-        order,
+        source: 'external',
+      },
+      orderLine: {
+        id: 1,
+        order: 1,
         type: 'item',
-      });
-      orderSelection = store.createRecord('order-selection', {
+      },
+      orderSelection: {
         id: 1,
-        order,
-        line: orderLine,
+        order: 1,
+        line: 1,
         quantity: 1,
-      });
-      orderSelectionQuestion = store.createRecord('order-selection-question', {
-        id: 1,
-        order,
-        selection: orderSelection,
-        text: 'answer',
-      });
-      orderSelectionQuestion2 = store.createRecord('order-selection-question', {
-        id: 2,
-        order,
-        selection: orderSelection,
-      });
-
-      fakeSave(order);
-      fakeSave(orderLine);
-      fakeSave(orderSelection);
-      fakeSave(orderSelectionQuestion);
-      fakeSave(orderSelectionQuestion2);
-    });
+      },
+      orderSelectionQuestions: [
+        {
+          id: 1,
+          order: 1,
+          selection: 1,
+          text: 'answer',
+        },
+        {
+          id: 2,
+          order: 1,
+          selection: 1,
+        },
+      ],
+    };
+
+    store.pushPayload(payload);
```

Comment:
Did the `fakeSave` re-implementation not work? Is the state of the records the same? Before `isNew` would of been `false` I think whereas now they are `true`, does this matter?

Reply:
It was working but this looked like a cleaner solution than overriding adapter methods. Do you think `isNew` is relevant since we're only testing relationships?

Reply:
@mrloop No, pushing payload means we obtained the record outside the adapter's realm, but since it has an ID (I think it is required for `pushPayload`, it is equivalent to querying the record the "normal way"

Reply:
We think using public APIs is safer, and `pushPayload` does exactly what the original intention was with the test setup

Reply:
@mrloop @JMilosevic810 pretty sure `isNew` was false previously as well.